src/App.js

```diff
@@ -1,24 +1,174 @@
-import logo from './logo.svg';
-import './App.css';
+import tasks from "./tasks";
+import "./App.css";
+import { useState } from "react";
+import { TaskType } from "./TaskType";
+import Modal from "./components/Modal";
+import Board from "./components/Board";
 
 function App() {
+  const [show, setShow] = useState(false);
+  const [plannedState, setPlannedState] = useState(
+    tasks.filter((t) => t.state === TaskType.Todo)
+  );
+  const [startedState, setStartedState] = useState(
+    tasks.filter((t) => t.state === TaskType.InProgress)
+  );
+  const [doneState, setDoneState] = useState(
+    tasks.filter((t) => t.state === TaskType.Done)
+  );
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [currTask, setCurrTask] = useState();
+  const [taskType, setTaskType] = useState();
+  const showModal = () => {
+    setShow(true);
+  };
+
+  const hideModal = () => {
+    setShow(false);
+    setIsEditing(false);
+  };
+  
+  /**
+   * Adds a task to the respective state
+   * @param {Object} task
+   */
+  const addTask = (task) => {
+    switch (task.state) {
+      case TaskType.Todo:
+        setPlannedState([...plannedState, task]);
+        break;
+
+      case TaskType.InProgress:
+        setStartedState([...startedState, task]);
+        break;
+
+      case TaskType.Done:
+        setDoneState([...doneState, task]);
+        break;
+      default:
+      // Do nothing
+    }
+  };
+  const getUpdatedTaskList = (updatedTask, taskList, setState) => {
+    const updatedTaskList = taskList.map(task => task.id === updatedTask.id ? updatedTask : task);
+    setState(updatedTaskList);
+  };
+
+  const updateTask = (updatedTask) => {
+    if (updatedTask.state === currTask.state) {
+      switch (updatedTask.state) {
+        case TaskType.Todo:
+          getUpdatedTaskList(updatedTask, plannedState, setPlannedState);
+          break;
+        case TaskType.InProgress:
+          getUpdatedTaskList(updatedTask, startedState, setStartedState);
+          break;
+        case TaskType.Done:
+          getUpdatedTaskList(updatedTask, doneState, setDoneState);
+          break;
+        default:
+          console.log("came in default");
+      }
+    } else {
+      // Delete task from old state => currTask.state
+      deleteTask(currTask.id, currTask.state);
+
+      // Add task to new state => updatedTask.state
+      addTask(updatedTask);
+    }
+  };
+
+  /**
+   * Edits the current task
+   * @param {Object} task
+   */
+  const editTask = (task) => {
+    setIsEditing(true);
+    setCurrTask(task);
+    setShow(true);
+  };
+
+  const addTaskType = (taskType) => {
+    setTaskType(taskType);
+  };
+
+  const deleteTask = (id, state) => {
+    let allTasks;
```

Comment:
Rename variable. This should not be `allTasks`.
```suggestion
    let tasks;
```
`tasks` could be a name you can use. Rename everywhere in this function.